plugins/destination/bigquery/client/write.go

```diff
@@ -58,3 +64,62 @@ func (c *Client) WriteTableBatch(ctx context.Context, table *schema.Table, resou
 
 	return nil
 }
+
+func (c *Client) getValueForBigQuery(col arrow.Array, i int) any {
+	switch v := col.(type) {
+	case *array.Struct:
+		m := map[string]bigquery.Value{}
+		fields := v.DataType().(*arrow.StructType).Fields()
+		for f, field := range fields {
+			m[field.Name] = c.getValueForBigQuery(v.Field(f), i)
+		}
+		return m
+	case *array.Map:
+		v2 := col.GetOneForMarshal(i)
+		b, _ := json.Marshal(v2)
+		return string(b)
+	case array.ListLike:
+		arr := col.(array.ListLike)
+		elems := make([]any, 0, arr.Len())
+		for j := 0; j < arr.Len(); j++ {
+			if arr.IsNull(j) {
+				continue
+			}
+			from, to := arr.ValueOffsets(j)
+			slc := array.NewSlice(arr.ListValues(), from, to)
+			for k := 0; k < int(to-from); k++ {
+				if slc.IsNull(k) {
+					// LIMITATION: BigQuery does not support null values in repeated columns.
+					// Therefore, these get stripped out here. In the future, perhaps we should support
+					// an option to use JSON instead of repeated columns for users who need to preserve
+					// the null values.
+					continue
+				}
+				elems = append(elems, c.getValueForBigQuery(slc, k))
+			}
+		}
+		return elems
+	case *array.MonthDayNanoInterval:
+		return v.Value(i)
+	case *array.DayTimeInterval:
+		return v.Value(i)
+	case *array.Duration:
+		return v.Value(i)
+	case *array.Timestamp:
+		unit := v.DataType().(*arrow.TimestampType).Unit
+		switch unit {
+		case arrow.Nanosecond:
+			t := v.Value(i).ToTime(arrow.Nanosecond)
+			format := "2006-01-02 15:04:05.999999"
+			return TimestampNanoseconds{
+				Timestamp:   t.Format(format),
+				Nanoseconds: t.Nanosecond() % 1000,
+			}
+		default:
+			return v.GetOneForMarshal(i)
+		}
+	case *types.JSONArray:
+		return v.ValueStr(i)
```

Comment:
Maybe using
```suggestion
		return v. GetOneForMarshal(i)
```
could be better?

Reply:
I had `GetOneForMarshal` before, but it doesn't work. Something about it being `json.RawMessage` seems to confuse BigQuery and it throws an error about it not being a record column